File: generator/app/templates/app/src/main/java/com/example/base/baseapp/repository/ExampleDataRepository.java
package com.example.base.baseapp.repository;

import com.example.base.baseapp.model.ExampleData;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface ExampleDataRepository {
    Single<List<ExampleData>> getAllExampleData();

    Completable insertExampleData(List<ExampleData> exampleDataList);
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/repository/ExampleDataRepository.kt
package com.example.base.baseapp.repository

import com.example.base.baseapp.model.ExampleData

import io.reactivex.Completable
import io.reactivex.Single

interface ExampleDataRepository {
    val allExampleData: Single<List<ExampleData>>

    fun insertExampleData(exampleDataList: List<ExampleData>): Completable
}
